File: backend/app/schemas/assessment.py
from pydantic import BaseModel, Field
from typing import List, Literal


class HeavyMetal(BaseModel):
    metal: str
    concentration: float = Field(..., gt=0)


class AssessmentRequest(BaseModel):
    heavy_metals: List[HeavyMetal]

    soil_ph: float = Field(..., ge=0, le=14)

    organic_matter: float = Field(
        ..., ge=0,
        description="Organic Matter (%)"
    )

    soil_texture: Literal[
        "sand",
        "silt",
        "clay"
    ]

    crop: str

File: frontend/src/types/assessment.ts
export interface ManualMetal {
    metal: string;
    concentration: number;
}

export interface AssessmentData {
    crop: string;

    soilPH: number;

    organicMatter: number;

    soilTexture: string;

    csvFile: File | null;

    manualMetals: ManualMetal[];
}